detection_rules/integrations.py

```diff
@@ -0,0 +1,127 @@
+        if "endpoint" in rule_integrations:
+            rule_integrations.remove("endpoint")
+
+        final_integration_manifests = dict()
+        for ints in rule_integrations:
+            integration_manifest = get_integration_packages(token=token, org="elastic", repo="package-storage",
+                                                            branch="production", folder=f"packages/{ints}")
+            final_integration_manifests.setdefault(ints, integration_manifest[ints])
+
+        manifest_file = gzip.open(manifest_file_path, "w+")
+        manifest_file_bytes = json.dumps(final_integration_manifests).encode("utf-8")
+        manifest_file.write(manifest_file_bytes)
+
+    @classmethod
+    def find_least_compatible_version(cls, package: str, integration: str,
+                                      current_stack_version: str, packages_manifest: dict) -> str:
+        """Finds least compatible version for specified integration based on stack version supplied"""
+        integration_manifests = packages_manifest[package]
+        least_compatible_version = None
+        compatible_versions = dict()
+
+        def compare_versions(int_ver: str, pkg_ver: str) -> bool:
+            """Compares integration and package version"""
+            pkg_major, pkg_minor = Version(pkg_ver)
+            int_major, int_minor = Version(int_ver)[:2]
+
+            if int(int_major) < int(pkg_major) or int(pkg_major) > int(int_major):
+                return(False)
+            compatible = Version(int_ver) <= Version(pkg_ver)
+            return(compatible)
+
+        for ver, manifest in integration_manifests.items():
+            kibana_compat_vers = re.sub(r"\>|\<|\=|\^", "", manifest["conditions"]["kibana.version"])
+            kibana_compat_vers = kibana_compat_vers.split(" || ")
+            bool_checks = list()
+            for kcv in kibana_compat_vers:
+                bool_checks.append(compare_versions(kcv, current_stack_version))
+            if any(bool_checks):
+                compatible_versions.setdefault(ver, True)
+
+        if len(list(compatible_versions.keys())) > 0:
+            compatible_version_list = list(compatible_versions.keys())
+            compatible_version_list.sort(key=Version)
+            least_compatible_version = compatible_version_list[0]
+        else:
+            raise Exception(f"no compatible version for integration {package}:{integration}")
+        return(least_compatible_version)
+
+
+def get_integration_packages(token: str, org: str, repo: str, branch: str, folder: str) -> None:
+    """Gets integration packages object containing versioned packages and manifest content"""
+    github = GithubClient(token)
+    client = github.authenticated_client
+    organization = client.get_organization(org)
+    repository = organization.get_repo(repo)
+    sha = get_sha_for_branch(repository, branch)
+    integration_manifest = get_integration_manifests(repository, sha, folder)
+    return(integration_manifest)
```

Comment:
```suggestion
    return integration_manifest
```

Reply:
Resolved these everywhere they existed. They cause no difference in the return but to make the code consistent, it makes sense to remove.